### ISSUE TITLE
Estimates

### ISSUE THREAD

**Issue by MohdSahal** [techgebra/polosys-books#23]
- [x] Crud
- [x] Preferences
- [x] Show stock details of below quantity of item; including expand view.(common)
- [x] Tax treatment and TRN update option in creation screen(common).
- [x] Show primary and secondary address of selected customers and give option to manage multiple addresses.(common-all sales modules)
- [x] Change POS  by changing of tax treatment in creation screen(common).
- [x] Re-arrange 'place of supply(pos)' column in creation screen.(common)
- [x] Item tax treatment, RCM based on tax treatment (non-gcc and out of scope).(common)
- [x] Sales person creation screen only need name and email(common).
- [x] Edit, view item details option of selected item in item table.(common)
- [x] Option to see recent transactions of an item in item table below item rate.(common)
- [x] Show additional information and clone option in each line of item and table headings.(common)
- [x] Barcode required to for each item in item related vouchers.
- [x] 'Add items in bulk' and 'add item header' option in add another line button.
- [x] Retainer invoice creation with percentage of estimates(to manage advance payment of customer) in estimates creation screen. retainer creation only after acceptance with popup and permission of user.
- [x] Show all mails related to such customer in estimates creation screen with add new contact person option. give option to mark mails need to sent to customer.
- [x] Change template option in creation screen.
- [x] In detailed view (draft stage): 1. clone function- submit button rename to 'save as draft'. 2.send mail/mark as sent button only need to functioning after approval of such estimates(need popup for submit to approval). 3. need share option through links(only need to functioning after sent mail/mark as sent). 4. Path to Estimate preferences(common).
- [x] Options in templates(common): 1.Edit templates. 2. update logo and address. 3. manage custom fields(path to custom fields). 4. Terms& conditions
- [x] Sales order/convert to invoice only possible after accepted such estimate. block and popup before acceptance of such estimates.
- [x] Comment& history option(common), keep all history of estimates with user, date and time.
- [x] In detailed view ('submit to approval' stage): 1.give option to convert back to draft(give option to enter reasons). 2. for reject need column for enter reason.
- [x] In detailed view ('rejected' status): 1. give convert option to draft with reason.
- [x] In detailed view ('approved' status): 1. Convert to draft with giving reason. 2. view approval details option
- [x] In detailed view('accepted status'): 1. create retainer invoice option 
- [x] Flag on template related to status of such estimates (template content changed based on flags).
- [x] In detailed view (invoiced status): 1. list of invoices and show count 2. show related retainers created from estimates.
- [x] Block delete function for invoiced or retainer invoice linked with an estimates.
- [x] Buttons in edit screen 1. in draft stage (save, save and send, save and submit, save and approve) 2. in pending for approval screen(save, save and sent, save and approval) 3. in reject screen(save, save and submit). 3. in approved screen (save, save and send) 4. in sent/accepted/declined/invoiced screen(save, save and send).
- [x] Need confirmation from user for multiple conversion from estimates.
- [x] Remove option 'convert to sales order' in invoiced status.
- [ ]

**Comment by MohamedRoshanPU** [comment]
- Share Functionality in detail view (pending)

**Comment by MohdSahal** [comment]
completed
